org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -41,7 +42,11 @@
 
 /**
  * Main entry point for accessors to run requests on the language servers, and some utilities
- * for manipulating the asynchronous response objects in streams
+ * for manipulating the asynchronous response objects in streams.
+ *
+ * Note that versioning support makes the executor classes stateful. Attempting to call request
+ * methods multiple times on the same executor object will throw an <code>IllegalStateExeception</code>.
+ * Either call <code>reset()</code> between intervening executions or <code>clone()</code> a fresh object.
```

Comment:
I do not think we need the `reset` method, for the number of times we will actually want to reuse executors, I think just creating a new object should be good enough.

Reply:
You are perhaps right but I can foresee getting the reverse complaint as well. Certainly it's more of a convenience thing than a necessity thing if clone/copy is implemented.

Reply:
I agree, but since adding is easier than removing, I think we should start small. It is safer :)